packages/components/src/filter-picker/index.js

```diff
@@ -143,7 +167,30 @@ class FilterPicker extends Component {
 		config.staticParams.forEach( ( param ) => {
 			update[ param ] = query[ param ];
 		} );
-		updateQueryString( update, path, persistedQuery );
+
+		// Remove all of this filter's params not associated witth the update while
+		// leaving any other params from any other filter an extension may have added.
+		this.getAllFilterParams().forEach( ( param ) => {
+			if ( ! update[ param ] ) {
+				// Explicitly give value of undefined so it can be removed from the query.
+				update[ param ] = undefined;
+			}
+		} );
+
+		// If the main filter is being set to anything but advanced, remove any advancedFilters.
+		if ( config.param === 'filter' && value !== 'advanced' ) {
```

Comment:
Rather than give the published filter components more knowledge of our usage, could we move this logic into the `client/` app?

It probably requires us to refactor the `FilterPicker` component to play nice with other instances - this type of behavior is based on the assumption that there is only one picker modifying the `filter` param (here and elsewhere in the components).

Reply:
Big +1 on this- great idea, @jeffstieler